google/cloud/aiplatform/training_job_base.py

```diff
@@ -0,0 +1,298 @@
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import models
+from google.cloud.aiplatform import datasets
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform import utils
+
+from google.cloud.aiplatform_v1beta1.services.pipeline_service import (
+    client as pipeline_service_client,
+)
+
+from google.cloud.aiplatform_v1beta1.types import io as gca_io
+from google.cloud.aiplatform_v1beta1.types import model as gca_model
+from google.cloud.aiplatform_v1beta1.types import (
+    training_pipeline as gca_training_pipeline,
+)
+from google.cloud.aiplatform_v1beta1.types import pipeline_state as gca_pipeline_state
+
+from google.rpc import code_pb2
+
+logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+_LOGGER = logging.getLogger(__name__)
+
+_PIPELINE_COMPLETE_STATES = set(
+    [
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_SUCCEEDED,
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_FAILED,
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_CANCELLED,
+        gca_pipeline_state.PipelineState.PIPELINE_STATE_PAUSED,
+    ]
+)
+
+class DatasetWithSplits():
+    def __init__(
+        self,
+        dataset: datasets.Dataset,
+        training_fraction_split: float,
+        validation_fraction_split: float,
+        test_fraction_split: float
+    ):
+        self.dataset=dataset
+        self.training_fraction_split=training_fraction_split
+        self.validation_fraction_split=validation_fraction_split
+        self.test_fraction_split=test_fraction_split
+
+
+class TrainingJob(base.AiPlatformResourceNoun):
+    client_class = pipeline_service_client.PipelineServiceClient
+    _is_client_prediction_client = False
+
+    def __init__(
+        self,
+        display_name: str,
+        container_uri: str,
+        requirements: Optional[Sequence[str]] = None,
+        model_serving_container_image_uri: Optional[str] = None,
+        model_serving_container_predict_route: Optional[str] = None,
+        model_serving_container_health_route: Optional[str] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        super().__init__(project=project, location=location, credentials=credentials)
+        self._display_name = display_name
+        self._container_uri = container_uri
+        self._requirements = requirements
+        self._project = project
+        self._credentials = credentials
+        self._model_serving_container_image_uri = model_serving_container_image_uri
+        self._model_serving_container_predict_route = (
+            model_serving_container_predict_route
+        )
+        self._model_serving_container_health_route = (
+            model_serving_container_health_route
+        )
+        self._gca_resource = None
+
+    def _create_managed_model(self, model_display_name: str) -> gca_model.Model:
```

Comment:
Encapsulated this logic into a function to emphasize the parameter dependencies to generate a managed model.

Reply:
Hm, seems like this isn't used by AutoML. I guess it's specific to Custom Training then? Will move there then

Reply:
I ended up having each subclass create their own `Model` and pass it in to the base class. This is due to the fact that each training job type has different requirements as defined in the yaml files.